### DISCUSSION TITLE
The material of the custom item I added cannot be displayed! MC 1.21.4

### DISCUSSION THREAD

**Discussion by ZephyrBD** [FabricMC/fabric-api#4326] [Mod Dev Support] [answered]
### Description

I added a custom_item according to the Fabirc docs and also added the material file according to the path, but the material is always lost in the game.
My code:
```
public class ModItems {
    public static final Item CUSTOM_ITEM = register("custom_item", Item::new, new Item.Settings().maxCount(64));
    public static Item register(String name, Function<Item.Settings, Item> factory, Item.Settings settings) {
        final RegistryKey<Item> registryKey = RegistryKey.of(RegistryKeys.ITEM, Identifier.of(FirstFabricMod.MOD_ID, name));
        return Items.register(registryKey, factory, settings);
    }
    public static void initialize() {
    }
}
```
custom_item.json
```
{
  "parent": "minecraft:item/generated",
  "textures": {
    "layer0": "firstfabricmod:item/custom_item"
  }
}
```
Flies Tree:
![image](https://github.com/user-attachments/assets/92ea5ae4-a8c6-4fc8-b61b-b90ecf25144f)

MC version:  1.21.4
Fabric version: 0.16.9
[latest.log](https://github.com/user-attachments/files/18243582/latest.log)

**Comment by cputnam-a11y** [comment]
You appear to have forgotten your item json file. As of 1.21.4 it is required in addition to the item model file. The 1.21.4 blog post describes some of how you can create it

**Reply by cputnam-a11y** [reply]
https://fabricmc.net/

**Reply by ZephyrBD** [reply]
So, could you please tell me where I might be able to access this blog?

**Reply by cputnam-a11y** [reply, marked as answer]
https://fabricmc.net/2024/12/02/1214.html

**Reply by ZephyrBD** [reply]
thanks，my network is not good，sorry
